### DISCUSSION TITLE
How to use internal APIs for custom commands

### DISCUSSION THREAD

**Discussion by NoelDeMartin** [cypress-io/cypress#27065] [Questions and Help]
I'm going through the guide to make [Custom Queries](https://docs.cypress.io/api/cypress-api/custom-queries), and there are a couple of examples using internal APIs like [cy.state](https://github.com/cypress-io/cypress/blob/89f0fb64650d1795a912773b5c0660d1aceea0bc/packages/driver/types/internal-types.d.ts#L24).

However, since I am writing my code in TypeScript, I can't use the internal types. Is there a know workaround for this? Like installing a separate `@types` package or something?

**Comment by aniketjivtode123** [comment]
Hey @NoelDeMartin, did you find any solution around this ?

**Reply by NoelDeMartin** [reply]
I didn't find any decent workaround, so I'm just [manually declaring the types I'm using](https://github.com/NoelDeMartin/aerogel/blob/main/packages/cypress/src/support/queries/index.ts#L24..L34) :/. For example, to declare `cy.state` I added this in my codebase:

```ts

declare global {
    namespace Cypress {
        interface Chainable {
            state(type: 'current'): Cypress.Command;
        }
    }
}
```
